### PR TITLE
feat: Inform about new devices (AR-2964)

### PR DESCRIPTION
**Pull request by borichellow**
<table>
<td>
  <a href="https://wearezeta.atlassian.net/browse/AR-2964" title="AR-2964" target="_blank">StoryAR-2964</a>  Get informed about newly added devices
  </td></table>
  <br />
 

# Waiting for kalium PR
https://github.com/wireapp/kalium/pull/1523

# What's new in this PR?

As a wire user I want to get informed if someone logged in from a new device into my account.

### Attachments (Optional)

photo_5454118852989404485_y

photo_5454118852989404484_y

### CODE REVIEW

**Comment by AndroidBob** [comment]
Build [1](https://10.10.124.134/job/wire-android-reloaded/job/PR-1487/1/) **failed**.

**Comment by AndroidBob** [comment]
Build [2](https://10.10.124.134/job/wire-android-reloaded/job/PR-1487/2/) **failed**.

File: app/src/main/kotlin/com/wire/android/ui/WireActivityViewModel.kt [hunk 1]
```diff
@@ -282,6 +313,16 @@ class WireActivityViewModel @Inject constructor(
         }
     }
 
+    fun dismissNewClientDialog() {
+        globalAppState = globalAppState.copy(newClientDialog = null)
+    }
+
+    fun switchAccount(userId: UserId) {
```
Comment (saleniuk) [inline app/src/main/kotlin/com/wire/android/ui/WireActivityViewModel.kt:264]: Would that be possible to switch account and open the `SelfDevices` for that switched account? Maybe by using deep links just like you are implementing right now for conversations and calls? That would be way better UX I think 😄

File: app/src/main/kotlin/com/wire/android/ui/WireActivityViewModel.kt [hunk 2]
```diff
@@ -465,10 +510,22 @@ sealed class CurrentSessionErrorState {
...
 }
 
+sealed class NewClientData(open val date: String, open val deviceInfo: String) {
+    data class CurrentUser(override val date: String, override val deviceInfo: String) : NewClientData(date, deviceInfo)
+    data class OtherUser(
+        override val date: String,
+        override val deviceInfo: String,
+        val userId: UserId,
+        val userName: String?,
+        val userHandler: String?
```
Comment (saleniuk) [inline app/src/main/kotlin/com/wire/android/ui/WireActivityViewModel.kt]: I think it's a typo and it's `userHandle` without the `r` 😄

File: app/src/main/kotlin/com/wire/android/ui/WireActivityViewModel.kt (hunk 1 above)
Comment (borichellow) [reply app/src/main/kotlin/com/wire/android/ui/WireActivityViewModel.kt:264]: yeah, from now we can achieve it easily :D

File: app/src/main/kotlin/com/wire/android/ui/WireActivityViewModel.kt (hunk 2 above)
Comment (borichellow) [reply app/src/main/kotlin/com/wire/android/ui/WireActivityViewModel.kt]: i really thought it's handler :D

**Comment by AndroidBob** [comment]
Build [5](https://10.10.124.134/job/wire-android-reloaded/job/PR-1487/5/) **failed**.

**Comment by AndroidBob** [comment]
Build [6](https://10.10.124.134/job/wire-android-reloaded/job/PR-1487/6/) **succeeded**.

The build produced the following APK's:
- [com.wire.android-vv4.0.2(5931847)(5931847)-staging-compat.apk](https://z-lohika.s3-eu-west-1.amazonaws.com/megazord/android/reloaded/staging/compat/com.wire.android-vv4.0.2(5931847)(5931847)-staging-compat.apk)

File: app/src/test/kotlin/com/wire/android/ui/WireActivityViewModelTest.kt [hunk 3]
```diff
@@ -569,6 +570,7 @@ class WireActivityViewModelTest {
             every { observeSyncStateUseCaseProviderFactory.create(any()).observeSyncState } returns observeSyncStateUseCase
             every { observeSyncStateUseCase() } returns emptyFlow()
             coEvery { observeIfAppUpdateRequired(any()) } returns flowOf(false)
+            coEvery { observeNewClients() } returns flowOf()
```
Comment (gongracr) [inline app/src/test/kotlin/com/wire/android/ui/WireActivityViewModelTest.kt:601]: Can we add a test verifying that the newly added client dialog will be shown if a new device is observed from the current session and other account?

**Comment by AndroidBob** [comment]
Build [7](https://10.10.124.134/job/wire-android-reloaded/job/PR-1487/7/) **succeeded**.

The build produced the following APK's:
- [com.wire.android-vv4.0.2(5941082)(5941082)-staging-compat.apk](https://z-lohika.s3-eu-west-1.amazonaws.com/megazord/android/reloaded/staging/compat/com.wire.android-vv4.0.2(5941082)(5941082)-staging-compat.apk)

**Review by gongracr — APPROVED** [review]
Thanks for the changes. Looking now much fresher 🥬 🧀 🍅

**Review by saleniuk — APPROVED** [review]

**Comment by AndroidBob** [comment]
Build [8](https://10.10.124.134/job/wire-android-reloaded/job/PR-1487/8/) **succeeded**.

The build produced the following APK's:
- [com.wire.android-vv4.0.2(5983873)(5983873)-staging-compat.apk](https://z-lohika.s3-eu-west-1.amazonaws.com/megazord/android/reloaded/staging/compat/com.wire.android-vv4.0.2(5983873)(5983873)-staging-compat.apk)

**Pull request merged**